app/controllers/api/v1/books_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      before_action :authenticate_api_v1_user!
+
+      def index
+        @books = Book.all
+        render_paginated @books, each_serializer: BookIndexSerializer
+      end
+
+      def show
+        @book = Book.find_by(id: params[:id])
```

Comment:
use `Book.find(params[:id])` use find_by when you want a search for other attributes

Reply:
but with find if record is not in DB, raise an error but with find_by returns nill

Reply:
rescue not found exceptions in application controller with the not_found method